### PR TITLE
s582 admin improvements

### PR DESCRIPTION
**Pull request by ProtoJazz**
I pulled the Newsletter settings into its own page. Turned out to involve more than I guessed it would. 
It will probably look better once there are more settings in there. 

This is the first PR for some general admin cleanup (https://github.com/tidusjar/PlexRequests.Net/issues/582) 

Figured Id start with a small chunk to make sure I'm doing it right.

### CODE REVIEW

**Comment by ProtoJazz** [comment]
oops. I keep opening it for master instead of

**Comment by tidusjar** [comment]
Looks good so far, the only thing we have to be concerned about is that since we are changing the shape of the settings, any existing values will be lost, so we would need to write some code to migrate the settings into the new shape if you get me

**Comment by ProtoJazz** [comment]
Alright, so sounds like I should make a OldToNew migration somewhere that moves the old values to the new and removes them from the old. Is there a good place to run stuff like that when the app starts?

**Comment by tidusjar** [comment]
There's a file in the Core project called Setup.cs.

There is some code to migrate in there.

**Comment by ProtoJazz** [comment]
Alright. I gave a shot at migration setup. 
I guess I have to leave the old settings in the PlexRequestsSettings file? Otherwise Im not sure how to transfer the old value.

File: PlexRequests.Core/SettingModels/PlexRequestSettings.cs [hunk 1]
```diff
@@ -58,7 +58,8 @@ public PlexRequestSettings()
         public bool Wizard { get; set; }
         public bool DisableTvRequestsByEpisode { get; set; }
         public bool DisableTvRequestsBySeason { get; set; }
-
+        //To Remove at some point? 
+        public bool SendRecentlyAddedEmail { get; set; }
```
Comment (tidusjar) [inline PlexRequests.Core/SettingModels/PlexRequestSettings.cs]: I usually apply the `[Obsolete]` attribute with a description. It add's build warnings which in a few versions time I can then remove :)

**Comment by ProtoJazz** [comment]
Fixed merge issues, fixed [Obsolete] attribute.

**Comment by tidusjar** [comment]
Hey! Can you fix the last bit of the merge issues?

**Comment by ProtoJazz** [comment]
Merged issues

**Comment by tidusjar** [comment]
Thanks!!!!!

**Comment by whitecollar17** [comment]
@tidusjar @ProtoJazz 

I'm now on the last dev build, but i got a error when i click newsletter in the admin page

> Error Details
> 
> Nancy.RequestExecutionException: Oh noes! ---> Nancy.ViewEngines.ViewNotFoundException: Unable to locate view 'NewsletterSettings'
> Currently available view engine extensions: sshtml,html,htm,cshtml,vbhtml
> Locations inspected: views/admin/Admin/NewsletterSettings-nl-NL,views/admin/Admin/NewsletterSettings,admin/Admin/NewsletterSettings-nl-NL,admin/Admin/NewsletterSettings,views/admin/NewsletterSettings-nl-NL,views/admin/NewsletterSettings,admin/NewsletterSettings-nl-NL,admin/NewsletterSettings,views/Admin/NewsletterSettings-nl-NL,views/Admin/NewsletterSettings,Admin/NewsletterSettings-nl-NL,Admin/NewsletterSettings,views/NewsletterSettings-nl-NL,views/NewsletterSettings,NewsletterSettings-nl-NL,NewsletterSettings
> Root path: C:\PlexRequests.net
> If you were expecting raw data back, make sure you set the 'Accept'-header of the request to correct format, for example 'application/json'
>    at Nancy.ViewEngines.DefaultViewFactory.GetRenderedView(String viewName, Object model, ViewLocationContext viewLocationContext)
>    at System.Dynamic.UpdateDelegates.UpdateAndExecute4[T0,T1,T2,T3,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3)
>    at CallSite.Target(Closure , CallSite , DefaultViewFactory , Object , Object , ViewLocationContext )
>    at Nancy.ViewEngines.DefaultViewFactory.RenderView(String viewName, Object model, ViewLocationContext viewLocationContext)
>    at System.Dynamic.UpdateDelegates.UpdateAndExecute4[T0,T1,T2,T3,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3)
>    at CallSite.Target(Closure , CallSite , IViewFactory , String , Object , ViewLocationContext )
>    at Nancy.Responses.Negotiation.ViewProcessor.Process(MediaRange requestedMediaRange, Object model, NancyContext context)
>    at System.Dynamic.UpdateDelegates.UpdateAndExecute4[T0,T1,T2,T3,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3)
>    at CallSite.Target(Closure , CallSite , IResponseProcessor , String , Object , NancyContext )
>    at Nancy.Responses.Negotiation.DefaultResponseNegotiator.NegotiateResponse(IEnumerable`1 compatibleHeaders, NegotiationContext negotiationContext, NancyContext context)
>    at Nancy.Responses.Negotiation.DefaultResponseNegotiator.CreateResponse(IList`1 compatibleHeaders, NegotiationContext negotiationContext, NancyContext context)
>    at Nancy.Responses.Negotiation.DefaultResponseNegotiator.NegotiateResponse(Object routeResult, NancyContext context)
>    at System.Dynamic.UpdateDelegates.UpdateAndExecute3[T0,T1,T2,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2)
>    at CallSite.Target(Closure , CallSite , IResponseNegotiator , Object , NancyContext )
>    at Nancy.Routing.DefaultRouteInvoker.<>c__DisplayClassa.b__6(Task`1 completedTask)
>    --- End of inner exception stack trace ---
>    at Nancy.NancyEngine.InvokeOnErrorHook(NancyContext context, ErrorPipeline pipeline, Exception ex)

**Comment by ProtoJazz** [comment]
Is this maybe an issue with my migration code?

**Comment by tidusjar** [comment]
Ah, on the view, NancyFX requires the view to be in the directory so the `Copy to Output Directory` on the view properties needs to be copy. You didn't see it in the debug build since when it's in debug it will always look at the compiled folders. I'll fix it

**Comment by tidusjar** [comment]
Also @ProtoJazz I've rewritten the way we do migrations, it's a lot better now. 

I've moved your migrations code to use the new way (See PlexRequests.Core.Migration.Migrations.Version195.cs)

**Comment by ProtoJazz** [comment]
@tidusjar awesome. Thanks. Didn't notice the copy to output thing at all

**Comment by tidusjar** [comment]
@ProtoJazz That's fine, I should have mentioned it! 

It would be good to get your feedback on the new way of migrations when you have time :)

**Comment by whitecollar17** [comment]
I test the last build. Last push fixed the problem (Y)

**Pull request merged**